yoticlient.go

```diff
@@ -63,6 +87,63 @@ func (client *Client) GetActivityDetails(token string) (ActivityDetails, []strin
 	return activityDetails, errStrings
 }
 
+func handleHTTPError(response *httpResponse, errorMessages ...map[int]string) error {
+	for _, handler := range errorMessages {
+		for code, message := range handler {
+			if code == response.StatusCode {
+				return fmt.Errorf(
+					message,
+					response.StatusCode,
+					response.Content,
+				)
+			}
+
+		}
+		if defaultMessage, ok := handler[-1]; ok {
+			return fmt.Errorf(
+				defaultMessage,
+				response.StatusCode,
+				response.Content,
+			)
+		}
+
+	}
+	return fmt.Errorf(
+		defaultUnknownErrorMessageConst,
+		response.StatusCode,
+		response.Content,
+	)
+}
+
+// MakeRequest is used by other yoti Packages to make requests using a single
+// common client object. Users should not use this function directly
+func (client *Client) MakeRequest(httpMethod, endpoint string, payload []byte, httpErrorMessages ...map[int]string) (responseData string, err error) {
```

Comment:
Is it possible to have `GetActivityDetails` and `performAmlCheck` make use of `MakeRequest`?